packages/webcert/src/feature/certificate/UnifiedEdit/UeInteger/UeInteger.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  CertificateDataElement,
+  ConfigUeInteger,
+  QuestionValidationTexts,
+  TextInput,
+  ValidationError,
+  ValueInteger,
+} from '@frontend/common'
+import { ValidationWrapper } from '@frontend/common/src/components/Inputs/DatePickerCustom/Styles'
+import * as React from 'react'
+import { useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import styled from 'styled-components'
+import { updateCertificateDataElement } from '../../../../store/certificate/certificateActions'
+import { getVisibleValidationErrors } from '../../../../store/certificate/certificateSelectors'
+
+const INVALID_NUMBER_PERIOD_ERROR = 'Ange ett värde mellan 0 och 100 %'
+
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+`
+const StyledTextInput = styled(TextInput)`
+  width: 45px;
+  padding: 4px 4px;
+  height: 35px;
+  text-align: center;
+  margin-right: 0.625em;
+`
+export interface Props {
+  question: CertificateDataElement
+  disabled: boolean
+}
+const UeInteger: React.FC<Props> = ({ question, disabled }) => {
+  const dispatch = useDispatch()
+  const questionValue = question.value as ValueInteger
+  const questionConfig = question.config as ConfigUeInteger
+  const [number, setNumber] = useState<string>('')
+  const validationErrors = useSelector(getVisibleValidationErrors(question.id))
+
+  useEffect(() => {
+    if (questionValue.value) {
+      setNumber(questionValue.value.toString())
+      isNumberValid(questionValue.value.toString())
+    }
+  }, [questionValue.value])
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === ' ') {
+      event.preventDefault()
+    }
+  }
+
+  const handleNumberOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value
+
+    if (/^[a-zA-Z]+$/.test(inputValue)) {
+      return null
+    }
+    setNumber(event.target.value.replace(/[^0-9-]/g, ''))
+
+    isNumberValid(inputValue)
+    dispatch(
+      updateCertificateDataElement({
+        ...question,
+        value: { ...questionValue, value: inputValue },
+      })
+    )
+  }
+
+  const isNumberValid = (number: string): ValidationError | undefined => {
```

Comment:
What does this do? Are you really using the returned object? 
0 and 100 should not be hard coded, you should use min and max values from config

Reply:
Maybe no longer needed? I will remove it :)